### PR TITLE
Skopeo: Use makefile to build

### PR DESCRIPTION
**Pull request by glensc**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Requires:
- https://github.com/containers/skopeo/pull/1507

### CODE REVIEW

**Comment by glensc** [comment]
Test build

```
brew install -s skopeo  -d  -v  --keep-tmp --HEAD
```

```
==> make bin/skopeo
CGO_CFLAGS="-I/usr/local/opt/gpgme/include -I/usr/local/Cellar/libassuan/2.5.5/include -I/usr/local/opt/libgpg-error/include" CGO_LDFLAGS="-L/usr/local/opt/gpgme/lib -lgpgme -L/usr/local/Cellar/libassuan/2.5.5/lib -lassuan -L/usr/local/opt/libgpg-error/lib -lgpg-error" GO111MODULE=on go build -mod=vendor  -ldflags '-X main.gitCommit=d9bc86fbe159da74aeb74a99f2c6e171ca828299 -X github.com/containers/image/v5/docker.systemRegistriesDirPath=/usr/local/etc/containers/registries.d -X github.com/containers/image/v5/signature.systemDefaultPolicyPath=/usr/local/etc/containers/policy.json -X github.com/containers/image/v5/pkg/sysregistriesv2.systemRegistriesConfPath=/usr/local/etc/containers/registries.conf -X github.com/containers/image/v5/internal/tmpdir.unixTempDirForBigFiles=/var/tmp ' -gcflags "" -tags "btrfs_noversion exclude_graphdriver_btrfs libdm_no_deferred_remove " -o bin/skopeo ./cmd/skopeo
```

**Comment by glensc** [comment]
LGTM

```
[~] ➔ brew uninstall skopeo
[~] ➔ brew install -s skopeo  --HEAD
==> Cloning https://github.com/lsm5/skopeo.git
Updating /Users/glen/Library/Caches/Homebrew/skopeo--git
==> Checking out branch config-file-prefix
Already on 'config-file-prefix'
Your branch is up to date with 'origin/config-file-prefix'.
HEAD is now at d9bc86fb Makefile: set CONTAINERSCONFDIR based on the OS
==> make PREFIX=/usr/local/Cellar/skopeo/HEAD-d9bc86f install-binary
==> Caveats
Bash completion has been installed to:
  /usr/local/etc/bash_completion.d
==> Summary
🍺  /usr/local/Cellar/skopeo/HEAD-d9bc86f: 6 files, 26.4MB, built in 30 seconds
```

```
➔ skopeo --version
skopeo version 1.5.2-dev commit: d9bc86fbe159da74aeb74a99f2c6e171ca828299
```

File: Formula/skopeo.rb [hunk 1]
```diff
@@ -4,6 +4,8 @@ class Skopeo < Formula
   url "https://github.com/containers/skopeo/archive/v1.5.1.tar.gz"
   sha256 "624fd87dd8de7623f8c19c09715dd6b37820101e605ff5951cc512cf50d067a1"
   license "Apache-2.0"
+#  head "https://github.com/containers/skopeo.git", branch: "main"
+  head "https://github.com/lsm5/skopeo.git", branch: "config-file-prefix"
```
Comment (georgettica) [inline Formula/skopeo.rb:8]: just to make it easier
```suggestion
head "https://github.com/lsm5/skopeo.git", branch: "config-file-prefix"
```

**Comment by glensc** [comment]
Still waiting for skopeo to merge related PR.

**Comment by glensc** [comment]
Still waiting for upstream:
- https://github.com/containers/skopeo/pull/1507#issuecomment-1008931663

**Comment by glensc** [comment]
Still waiting upstream

**Comment by carlocab** [comment]
This branch is pretty stale by now. I suggest opening a new PR once whatever this is waiting for upstream is resolved.

**Pull request closed without merging**